lib/components/molecules/Table/Table.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable flowtype/no-types-missing-file-annotation */
+import React, { cloneElement, Children } from 'react';
+import styled from 'styled-components';
+
+import styles from './Table.style';
+// eslint-disable-next-line flowtype/no-types-missing-file-annotation
+import type { Props, ColumnProps } from './types';
+
+// eslint-disable-next-line flowtype/no-types-missing-file-annotation
+const Table = ({ children, data, className, vertical, ...props }: Props) => {
+  const Head = () => {
+    return (
+      <thead>
+        <tr role="row">
+          {Children.map(children, column => (
+            <th key={column.props.name} role="columnheader">
+              {cloneElement(column, { intent: 'header' })}
+            </th>
+          ))}
+        </tr>
+      </thead>
+    );
+  };
+  // eslint-disable-next-line no-empty-pattern
+  const Tbody = ({}: Props) => {
+    return (
+      <tbody>
+        {data.map(item => (
+          <tr key={item.id} role="row">
+            {Children.map(children, column => (
+              <td role="cell">
+                {cloneElement(column, {
+                  intent: 'cell',
+                  value: item[column.props.dataMappingKey],
+                })}
+              </td>
+            ))}
+          </tr>
+        ))}
+      </tbody>
+    );
+  };
+
+  const VerticalHeader = () => {
+    return Children.map(children, column => (
+      <tr key={column.props.name}>
+        <th role="columnheader">{cloneElement(column, { intent: 'header' })}</th>
```

Comment:
Why are you using cloneElement? the same can be done with Children

Here is more explanation 
https://stackoverflow.com/questions/37521798/when-should-i-be-using-react-cloneelement-vs-this-props-children/50441271#50441271